NameLayerMain/src/vg/civcraft/mc/namelayer/command/commands/UpdateName.java

```diff
@@ -0,0 +1,97 @@
+package vg.civcraft.mc.namelayer.command.commands;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.UUID;
+
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+import vg.civcraft.mc.namelayer.NameAPI;
+import vg.civcraft.mc.namelayer.NameLayerPlugin;
+import vg.civcraft.mc.namelayer.command.PlayerCommandMiddle;
+import vg.civcraft.mc.namelayer.misc.NameFetcher;
+
+public class UpdateName extends PlayerCommandMiddle {
+
+	private Map<UUID, String> newNames;
+
+	public UpdateName(String name) {
+		super(name);
+		setIdentifier("nlun");
+		setDescription("Updates your name on this server to the one your minecraft account currently has");
+		setUsage("/nlun [CONFIRM]");
+		setArguments(0, 1);
+		newNames = new TreeMap<UUID, String>();
+	}
+
+	public boolean execute(CommandSender sender, String[] args) {
+		if (!(sender instanceof Player)) {
+			sender.sendMessage(ChatColor.LIGHT_PURPLE
+					+ "NO NO NO NO NO NO NO NO");
+			return true;
+		}
+		final Player p = (Player) sender;
+		final UUID uuid = p.getUniqueId();
+		String oldName = NameAPI.getCurrentName(uuid);
+		
+		if (NameLayerPlugin.getGroupManagerDao().hasChangedNameBefore(uuid)) {
+			p.sendMessage(ChatColor.RED + "You already changed your name");
+			return true;
+		}
+
+		if (args.length == 0) {
+			Bukkit.getScheduler().runTaskAsynchronously(
+					NameLayerPlugin.getInstance(), new Runnable() {
+						@Override
+						public void run() {
+							List<UUID> uuids = new LinkedList<UUID>();
+							uuids.add(uuid);
+							NameFetcher fetcher = new NameFetcher(uuids);
+							Map<UUID, String> fetchedNames = null;
+							try {
+								fetchedNames = fetcher.call();
+							} catch (Exception e) {
+								e.printStackTrace();
+							}
+							String newName = fetchedNames.get(uuid);
+							if (newName == null) {
+								p.sendMessage(ChatColor.RED + "An error occured. Try again later");
```

Comment:
I'm guessing you need to either `return;` here or wrap the next few statements in an `else` clause.